src/components/results/TableByQuestion.js

```diff
@@ -0,0 +1,189 @@
+   */
+  const comparator = (e1, e2) => {
+    if (e2 < e1) {
+      return -1;
+    }
+    if (e2 > e1) {
+      return 1;
+    }
+    return 0;
+  };
+
+  /**
+   * Helper function to get the correct comparator depending on whether we are sorting ascending or descending
+   *
+   * @param {string} order The order for which we want to get the comparator
+   * @returns {{(string, string): number}} The comparator corresponding to the required order
+   */
+  const getComparator = (order) =>
+    order === Order.DESC
+      ? (a, b) => comparator(a, b)
+      : (a, b) => -comparator(a, b);
+
+  /**
+   * Helper function to invert the current order when clicking on sorted column header
+   */
+  const handleRequestSort = () => {
+    const isAsc = order === 'asc';
+    setOrder(isAsc ? Order.DESC : Order.ASC);
+  };
+
+  return (
+    <Box sx={{ mb: 8 }}>
+      <Stack direction={'column'} spacing={4}>
+        <Typography variant={'h5'} component={'h5'}>
+          {question.data.question}
+        </Typography>
+        <TableContainer component={Paper}>
+          <Table>
+            <TableHead>
+              <TableRow>
+                <TableCell align={"left"}>
+                  <TableSortLabel
+                    active={true}
+                    direction={order ?? 'asc'}
+                    onClick={handleRequestSort}
+                  >
+                    {'User'}
+                    {
+                      <Box component="span" sx={visuallyHidden}>
+                        {order === Order.DESC
+                          ? 'sorted descending'
+                          : 'sorted ascending'}
+                      </Box>
+                    }
+                  </TableSortLabel>
+                </TableCell>
+                <TableCell align={"left"}>Answer</TableCell>
+                <TableCell align={"left"}>Date</TableCell>
+                <TableCell align={"left"}>Correct</TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {userList.sort(getComparator(order)).map((userId) => (
+                <TableRow
+                  key={userId}
+                  sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
+                >
+                  <TableCell component="th" scope="row" align="left">
+                    {userId}
+                  </TableCell>
+                  <TableCell align="left" sx={{ maxWidth: 350 }}>
+                    {getResponseDataForUserId(userId)}
+                  </TableCell>
+                  <TableCell align="left">
+                    {getResponseDateForUserId(userId)}
+                  </TableCell>
+                  <TableCell align="left">
+                    {computeCorrectness(
+                      getResponseForUserId(userId)?.data,
+                      question.data
+                    ) ? (
+                      <CheckCircleOutlined color={'success'} />
```

Comment:
You don't need `{ }` if it's a string.

Reply:
Ok, thanks for pointing that out, I'll be more careful not to use them for string.